Feature: configurable frame interval

gozmanyoni opened the pull request:
* Allowing to configure frameInterval
* Fixing dongoe connect/plugged state handling in example

steelbrain reviewed (commented):
Would it be possible to detect whether an Android or an iOS device has been connected and adapt the frame interval accordingly?

I’m thinking of the scenario where you want to set it up once and forget it and want your setup to work with both iOS and Android?

gozmanyoni commented:
It should be possible - we get the phone type on the plugged event if I recall

gozmanyoni commented:
@steelbrain done - I moved frameInterval out of the driver to the handler layer (Don't want a driver interface to have too many responsibilities)

And added a config section per phone type (also added an enum) to set frameInterval. CarPlay has it, AA does not - and consumers can override that

steelbrain commented on src/web/CarplayWeb.ts:
```diff
@@ -66,6 +67,17 @@ export default class CarplayWeb {
         this.clearPairTimeout()
+
+        const phoneTypeConfg = this._config.phoneConfig[message.phoneType]
+        if (phoneTypeConfg?.frameInterval) {
```
Should we check for `this._frameInterval` being null here in addition to `phoneTypeConfg?.frameInterval != null`?

gozmanyoni replied:
Technically there should never be more than 1 plugged event - but we don't control that - so I think it's good to check - or perhaps always clear and reset the interval

steelbrain reviewed (commented):
Tried this with AA and Carplay, works well. Thank you!

steelbrain reviewed (approved):
🚢
